signac/common/config.py

```diff
@@ -15,18 +15,28 @@
 
 logger = logging.getLogger(__name__)
 
-DEFAULT_FILENAME = ".signacrc"
-CONFIG_FILENAMES = [DEFAULT_FILENAME, "signac.rc"]
-HOME = os.path.expanduser("~")
-CONFIG_PATH = [HOME]
-FN_CONFIG = os.path.expanduser("~/.signacrc")
+PROJECT_CONFIG_FN = os.path.join(".signac", "config")
+USER_CONFIG_FN = os.path.expanduser(os.path.join("~", ".signacrc"))
+
+
+def _get_project_config_fn(root):
+    return os.path.abspath(os.path.join(root, PROJECT_CONFIG_FN))
+
+
+def _get_config_dirname(fn):
+    # TODO: Is the string replace a sufficiently portable solution? We could
+    # use pathlib's `.paths` attribute, but that has significant performance
+    # implications: this takes ~200 ns, whereas
+    # os.path.join(*(pathlib.PosixPath(fn).parts[:-2])) takes ~6 us, almost a
+    # 20x slowdown. On the other hand, loading a project should not be a
+    # frequent operation that leading to bottlenecks.
+    return fn.replace(os.path.sep + PROJECT_CONFIG_FN, "")
```

Comment:
We use `os.sep` elsewhere, so I'd recommend that as a canonical name. It's [imported](https://github.com/python/cpython/blob/9539400390494f4930c245b3f98453592f4a1a8c/Lib/os.py#L96) from `os.path`, though. 🤷
```suggestion
    return fn.replace(os.sep + PROJECT_CONFIG_FN, "")
```

Reply:
Sure